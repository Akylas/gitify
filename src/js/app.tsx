import * as React from 'react';
import {
  Redirect,
  HashRouter as Router,
  Route,
  Switch,
} from 'react-router-dom';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';

import configureStore from './store/configureStore';
import Loading from './components/loading';
import Sidebar from './components/sidebar';

import EnterpriseLoginRoute from './routes/enterprise-login';
import LoginRoute from './routes/login';
import NotificationsRoute from './routes/notifications';
import SettingsRoute from './routes/settings';

// Store
export const store = configureStore();

export const PrivateRoute = ({ component: Component, ...rest }) => {
  // @ts-ignore
  const authReducer = store.getState().auth;
  const isAuthenticated =
    authReducer.get('token') !== null ||
    authReducer.get('enterpriseAccounts').size > 0;

  return (
    <Route
      {...rest}
      render={props =>
        isAuthenticated ? (
          <Component {...props} />
        ) : (
          <Redirect
            to={{ pathname: '/login', state: { from: props.location } }}
          />
        )
      }
    />
  );
};

const theme = {
  primary: '#555B6E',
  success: '#69B578',
  info: '#8BA9C6',
  warning: '#FCAA67',
  danger: '#B7524F',

  grayLighter: '#f9fafa',
  grayLight: '#eceeef',

  purple: '#555b6e',
  purpleDark: '#3f4351',
};

export const App = () => {
  return (
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <Router>
          <div className="wrapper">
            <Loading />
            <Sidebar />

            <Switch>
              <PrivateRoute path="/" exact component={NotificationsRoute} />
              <PrivateRoute path="/settings" exact component={SettingsRoute} />
              <Route path="/login" component={LoginRoute} />
              <Route path="/enterpriselogin" component={EnterpriseLoginRoute} />
            </Switch>
          </div>
        </Router>
      </ThemeProvider>
    </Provider>
  );
};
