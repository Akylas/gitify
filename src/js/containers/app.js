import React from 'react';
import { connect } from 'react-redux';

import { fetchNotifications } from '../actions';
import Sidebar from '../components/sidebar';
import NetworkStatus from '../components/network-status';

export class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      networkConnected: navigator.onLine
    };
  }

  componentDidMount() {
    const self = this;
    window.addEventListener('offline', function(e) { self.handleNetworkStatus(); });
    window.addEventListener('online', function(e) { self.handleNetworkStatus(); });
  }

  componentWillUnmount() {
    window.removeEventListener('offline', this.handleNetworkStatus);
    window.removeEventListener('online', this.handleNetworkStatus);
  }

  handleNetworkStatus() {
    if (navigator.onLine) {
      this.setState({
        networkConnected: true
      });
      this.props.fetchNotifications();
    } else {
      this.setState({
        networkConnected: false
      });
    }
  }

  render() {
    if (!this.state.networkConnected) {
      return <NetworkStatus />;
    }

    return (
      <div>
        <Sidebar
          location={this.props.location}
          toggleSearch={() => this.toggleSearch()} />
        {this.props.children}
      </div>
    );
  };
};

export default connect(null, { fetchNotifications })(App);
