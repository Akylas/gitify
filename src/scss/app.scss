/* @group Colors */

$brand-purple: #555b6e;
$brand-purple-dark: darken($brand-purple, 10%);

$white: #fff;
$gray: #55595c;
$gray-dark: #373a3c;

$gray-light: darken($white, 10%);
$gray-darker: darken($gray, 5%);

/* @end Colors */

/* @group Bootstrap Overrides */

$sidebar-width: 100px;

$brand-primary: #555b6e;
$brand-success: #69b578;
$brand-info: #8ba9c6;
$brand-warning: #fcaa67;
$brand-danger: #b7524f;

@import '../../node_modules/bootstrap/scss/bootstrap.scss';

/* @end Bootstrap Overrides */
