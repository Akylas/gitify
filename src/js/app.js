import '../scss/app.scss';

import React from 'react';
import ReactDOM from 'react-dom';
import {
  Redirect,
  HashRouter as Router,
  Route,
  Switch,
} from 'react-router-dom';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';

const { ipcRenderer } = require('electron');

import { toggleSettingsModal } from './actions';
import configureStore from './store/configureStore';
import Sidebar from './components/sidebar';
import Loading from './components/loading';
import LoginPage from './routes/login';
import NotificationsRoute from './routes/notifications';
import EnterpriseLoginPage from './routes/enterprise-login';
import SettingsModal from './components/settings-modal';

// Store
const store = configureStore();

ipcRenderer.on('toggle-settings', () => {
  store.dispatch(toggleSettingsModal());
});

export const PrivateRoute = ({ component: Component, ...rest }) => {
  const authReducer = store.getState().auth;
  const isAuthenticated =
    authReducer.get('token') !== null ||
    authReducer.get('enterpriseAccounts').size > 0;

  return (
    <Route
      {...rest}
      render={props =>
        isAuthenticated ? (
          <Component {...props} />
        ) : (
          <Redirect
            to={{ pathname: '/login', state: { from: props.location } }}
          />
        )
      }
    />
  );
};

const theme = {
  primary: '#555B6E',
  success: '#69B578',
  info: '#8BA9C6',
  warning: '#FCAA67',
  danger: '#B7524F',

  grayLighter: '#f9fafa',
  grayLight: '#eceeef',

  purple: '#555b6e',
  purpleDark: '#3f4351',
};

ReactDOM.render(
  <Provider store={store}>
    <ThemeProvider theme={theme}>
      <Router>
        <div className="wrapper">
          <Loading />
          <SettingsModal />
          <Sidebar />

          <Switch>
            <PrivateRoute path="/" exact component={NotificationsRoute} />
            <Route path="/login" component={LoginPage} />
            <Route path="/enterpriselogin" component={EnterpriseLoginPage} />
          </Switch>
        </div>
      </Router>
    </ThemeProvider>
  </Provider>,
  document.getElementById('gitify')
);
