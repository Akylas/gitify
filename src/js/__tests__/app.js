import React from 'react';  // eslint-disable-line
import ReactDOM from 'react-dom';

jest.mock('../routes');
jest.mock('../store/configureStore', () => {
  return () => ({
    getState: () => {
      auth: {
        get: () => null;
      }
    },
    subscribe: () => {},
    dispatch: () => {},
  });
});

describe('app.js', function () {
  it('should render the things', function () {
    ReactDOM.render = jest.fn();
    require('../app');

    expect(ReactDOM.render).toHaveBeenCalledTimes(1);
  });
});
