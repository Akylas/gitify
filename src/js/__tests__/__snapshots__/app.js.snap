// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app.js should test the PrivateRoute component (authenticated) 1`] = `null`;

exports[`app.js should test the PrivateRoute component (non authenticated) 1`] = `null`;
